deploy/ssh.sh

```diff
@@ -0,0 +1,205 @@
+  # If provided then CA intermediate certificate will be copied to provided filename.
+  if [ -n "$ACME_DEPLOY_SSH_CAFILE" ]; then
+    Le_Deploy_ssh_cafile="$ACME_DEPLOY_SSH_CAFILE"
+    _savedomainconf Le_Deploy_ssh_cafile "$Le_Deploy_ssh_cafile"
+  fi
+  if [ -n "$Le_Deploy_ssh_cafile" ]; then
+    # backup file we are about to overwrite.
+    _cmdstr="$_cmdstr cp $Le_Deploy_ssh_cafile $_backupdir ;"
+    # copy new certificate into file.
+    _cmdstr="$_cmdstr echo \"$(cat "$_cca")\" > $Le_Deploy_ssh_cafile ;"
+    _info "will copy CA file to remote file $Le_Deploy_ssh_cafile"
+  fi
+
+  # FULLCHAIN is optional.
+  # If provided then fullchain certificate will be copied to provided filename.
+  if [ -n "$ACME_DEPLOY_SSH_FULLCHAIN" ]; then
+    Le_Deploy_ssh_fullchain="$ACME_DEPLOY_SSH_FULLCHAIN"
+    _savedomainconf Le_Deploy_ssh_fullchain "$Le_Deploy_ssh_fullchain"
+  fi
+  if [ -n "$Le_Deploy_ssh_fullchain" ]; then
+    # backup file we are about to overwrite.
+    _cmdstr="$_cmdstr cp $Le_Deploy_ssh_fullchain $_backupdir ;"
+    # copy new certificate into file.
+    _cmdstr="$_cmdstr echo \"$(cat "$_cfullchain")\" > $Le_Deploy_ssh_fullchain ;"
+    _info "will copy full chain to remote file $Le_Deploy_ssh_fullchain"
+  fi
+
+  # REMOTE_CMD is optional.
+  # If provided then this command will be executed on remote host.
+  # A 2 second delay is inserted to allow system to stabalize after
+  # executing a service stop.
+  if [ -n "$ACME_DEPLOY_SSH_REMOTE_CMD" ]; then
+    Le_Deploy_ssh_remote_cmd="$ACME_DEPLOY_SSH_REMOTE_CMD"
+    _savedomainconf Le_Deploy_ssh_remote_cmd "$Le_Deploy_ssh_remote_cmd"
+  fi
+  if [ -n "$Le_Deploy_ssh_remote_cmd" ]; then
+    if [ -n "$Le_Deploy_ssh_service_stop" ]; then
+      _cmdstr="$_cmdstr sleep 2 ;"
+    fi
+    _cmdstr="$_cmdstr $Le_Deploy_ssh_remote_cmd ;"
+    _info "Will execute remote command $Le_Deploy_ssh_remote_cmd"
+  fi
+
+  # SERVICE_START is optional.
+  # If provided then this command will be executed on remote host.
+  # A 2 second delay is inserted to allow system to stabalize after
+  # executing a service stop or previous command.
+  if [ -n "$ACME_DEPLOY_SSH_SERVICE_START" ]; then
+    Le_Deploy_ssh_service_start="$ACME_DEPLOY_SSH_SERVICE_START"
+    _savedomainconf Le_Deploy_ssh_service_start "$Le_Deploy_ssh_service_start"
+  fi
+  if [ -n "$Le_Deploy_ssh_service_start" ]; then
+    if [ -n "$Le_Deploy_ssh_service_stop" ] || [ -n "$Le_Deploy_ssh_remote_cmd" ]; then
+      _cmdstr="$_cmdstr sleep 2 ;"
+    fi
+    _cmdstr="$_cmdstr $Le_Deploy_ssh_service_start ;"
+    _info "Will start remote service with command $Le_Deploy_ssh_remote_cmd"
+  fi
+
+  if [ -z "$_cmdstr" ]; then
+    _err "No remote commands to excute. Failed to deploy certificates to remote server"
+    return 1
+  else
+    # something to execute.
+    # run cleanup on the backup directory, erase all older than 180 days.
+    _cmdstr="find $_backupprefix* -type d -mtime +180 2>/dev/null | xargs rm -rf ; $_cmdstr"
```

Comment:
No,  we should not use any more command if it's not really necessary.
You never know what the remove system is, and what command is missing from the remote system.

Let me show you an example.

I recently add a `_hex_dump()` function, first at https://github.com/Neilpang/acme.sh/commit/5f8daeeb6d987687ac896c128c327c2096b02009

It was firstly implemented by `od` command.   Since `od` is one the core utils of GNU, and I tested all the platforms we have here: https://github.com/Neilpang/acmetest,  including solaris, freebsd and Mac.

I thought it should be safe.
But soon, it gave me a bug:
https://github.com/Neilpang/acme.sh/issues/579

It was in a router, the `od` command is missing. unbelievable.

And soon again, another bug:
https://github.com/Neilpang/acme.sh/issues/595

It was in busybox, the `od` command is also missing.

I fixed a lot, and now the current version is:  https://github.com/Neilpang/acme.sh/pull/600/commits/4e4a6d83973fa0580ef8b2f1d7d0312fdd671543

I never thought the result when I first started function.


So, how about `xargs/find` ?   do you think it's more portable than `od` ?  I don't believe.

So, never use external command if not necessary.  That would make our code more safe and portable.


As for you script, please see: https://github.com/Neilpang/acme.sh/blob/master/acme.sh#L3343

we already have such `for/basename/_math/[ -ge ]` combination.